### ISSUE TITLE
Replace deprecated `@asyncio.coroutine` with `async`/`await`

### ISSUE THREAD

**Issue by uSpike** [pexpect/pexpect#677]
`asyncio.coroutine` is deprecated since python 3.8 and will be removed in 3.10 (coming this year): https://docs.python.org/3/library/asyncio-task.html#asyncio.coroutine

Coroutine functions should be migrated to using async/await syntax, which would mean requiring python 3.5 or newer, which itself went EOL on 2017-08-08.

**Comment by swt2c** [comment]
It didn't end up getting removed in 3.10, but it has been removed in 3.11: https://bugs.python.org/issue43216

It would be good to get this fixed.  :-)

**Comment by djuarezg** [comment]
May I presume the new fixed release could be published now that https://github.com/pexpect/pexpect/pull/764 and https://github.com/pexpect/pexpect/issues/752 have been done?

**Comment by StanHatko** [comment]
Can you please publish a release with this bug fixed, so that it works properly in Python 3.11?

**Comment by gsalvatella** [comment]
Please we should publish this. It's been years since the [fix](https://github.com/pexpect/pexpect/commit/70d735584d9420caf3f7bc50847dc80a28a1d575) and even Debian had to [patch](https://salsa.debian.org/python-team/packages/pexpect/-/blob/master/debian/patches/0004-python3.11.patch) pexpect because the fix hasn't been released.

**Comment by dluyer** [comment]
Agreed: this is broken in Python 3.11 so we should release with the fix.  Let's work out if there's anything else 'coming soon' or other reason to hold off on a release right now.

**Comment by StanHatko** [comment]
Could you please release the updated 4.9.0 version (which includes this bug fix) for conda as well (URLs: https://anaconda.org/anaconda/pexpect and https://anaconda.org/conda-forge/pexpect)? That way `conda install pexpect` will be for the updated version with this bug fixed.

**Comment by jquast** [comment]
Although I have permission to release to pypi, I don't believe that I have any permission to do anything about conda, I am not an anaconda package maintainer or contributor, there must be a better place to make the request to conda-forge to release their pexpect 4.8.0 package to 4.9.0?

**Comment by StanHatko** [comment]
The issue is fixed now in 4.9.0 (updated in conda-forge now as well https://github.com/conda-forge/pexpect-feedstock/pull/29), so I think this issue can be closed.
